src/aiperf/common/config/user_config.py

```
@@ -54,6 +59,34 @@ def _is_localhost_url(url: str) -> bool:
+    normalized_url = url.strip()
+    if not normalized_url:
+        raise ValueError("--otel-url cannot be empty.")
+
+    if not normalized_url.startswith(("http://", "https://")):
+        normalized_url = f"http://{normalized_url}"
+
+    parsed = urlparse(normalized_url)
+    if not parsed.scheme or not parsed.netloc:
+        raise ValueError(
+            f"Invalid --otel-url value: {url!r}. Expected host[:port] or a full URL."
+        )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)